nixos/modules/installer/tools/nixos-rebuild.sh

```diff
@@ -341,7 +344,7 @@ fi
 
 # First build Nix, since NixOS may require a newer version than the
 # current one.
-if [ -n "$rollback" -o "$action" = dry-build ]; then
+if [ -n "$rollback" -o -n "$generation" -o "$action" = dry-build ]; then
```

Comment:
I know this has been here before (for rollback), but why do we use a new nix version to switch to an *old* config?

Reply:
I mean ideally we would actually use the *old* nix (the one that was used to build the *old* system). We can't really do that due to global state like the nix db I guess, but still using an even newer nix feels wrong.
People might want to rollback, because an update broke important part of their system (internet connection, maybe something with the filesystem tooling etc.) so we should probably do as little as possible to get back to the old state?

Reply:
I totally get your point. You could pass `--no-build-nix` with the current implementation to avoid this. I don't know if that should be the default behaviour...

Reply:
Schema upgrades in nix database or the new (experimental) nix profiles could potentially be a problem.

Reply:
Did I get it right: It should be removed for both `--generation` and `--rollback`?

Reply:
In my opinion yes. I can see no reason to update nix on such an operation.